include/caf/click_detector.rst

```diff
@@ -0,0 +1,79 @@
+.. _caf_click_detector:
+
+CAF: Click detector module
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |click_detector| of the :ref:`lib_caf` (CAF) is responsible for generating a ``click_event`` when a known type of click is recorded for the button defined in the module configuration.
+
+Click type
+**********
+
+The module records the following click types:
+
+* :c:enumerator:`CLICK_SHORT` - Button pressed and released after short time.
+* :c:enumerator:`CLICK_NONE` - Button pressed and held for a period of time that is too long for :c:enumerator:`CLICK_SHORT`, but too short for :c:enumerator:`CLICK_LONG`.
+* :c:enumerator:`CLICK_LONG` - Button pressed and held for a long period of time.
+* :c:enumerator:`CLICK_DOUBLE` - Two sequences of the button press and release in a short time interval.
+
+The exact values of time intervals for click types are defined in the :file:`subsys/caf/modules/click_detector.c` file.
+
+Configuration
+*************
+
```

Comment:
Configuration was intended to start with list of options required for making the module work. See [buttons](https://github.com/nrfconnect/sdk-nrf/blob/master/include/caf/buttons.rst) for reference.

Reply:
There are no such options in this case. Only the module that submits button events is needed here (described in first sentence).

Reply:
There are. Summon @greg-fer.

```
To use the module, you must enable the following Kconfig options:

* :option:`CONFIG_CAF_CLICK_DETECTOR` - This option enables the click detector module.
* :option:`CONFIG_CAF_BUTTON_EVENTS` - This option enables the button events which are required for click detector module to work.

Reply:
Ok, changed. Previously the Kconfig option that enabled the module was added as a final step of configuration.
Sorry for misunderstanding.